src/services/RecordStreamProcessor.ts

```diff
@@ -0,0 +1,110 @@
+import { ValidationHelper } from "../utils/ValidationHelper";
+import { personalIdentityUtils } from "../utils/PersonalIdentityUtils";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { ExtSessionEvent, SessionEvent } from "../models/SessionEvent";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { buildGovNotifyEventFields } from "../utils/GovNotifyEvent";
+import { EnvironmentVariables } from "./EnvironmentVariables";
+import { ServicesEnum } from "../models/enums/ServicesEnum";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { IPRService } from "./IPRService";
+import { AppError } from "../utils/AppError";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { Constants } from "../utils/Constants";
+
+export class RecordStreamProcessor {
+  private static instance: RecordStreamProcessor;
+  private readonly logger: Logger;
+  private readonly metrics: Metrics;
+  private readonly validationHelper: ValidationHelper;
+  private readonly iprService: IPRService;
+  private readonly environmentVariables: EnvironmentVariables;
+
+  constructor(logger: Logger, metrics: Metrics) {
+    this.logger = logger;
+    this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.STREAM_PROCESSOR_SERVICE);
```

Comment:
We should update the STEAM_PROCESSOR_SERVICE  to RECORD_STREAM_PROCESSOR in the enum else will be confusing.

Reply:
Done :)